### PR TITLE
fix(ios): hide/show indicator once orientation changed

### PR DESCRIPTION
**Pull request by troZee**
# Summary

When orientation changed, page indicator did not behave in correct way. Instead of be hidden, it was displayed on the screen. Initially in portrait mode indicator was not displayed, but in landscape mode, when orientation changed, indicator was displayed. It should be hidden all the time, unless `showPageIndicator` is set to `true`. 

I noticed, in curl mode `setHidden` does not work. I could find any reliable solution for it and I hacked a little bit VP. 

Closes https://github.com/react-native-community/react-native-viewpager/issues/182

## Test Plan

* set ` transitionStyle="curl"`
* rotate the screen 
* check, if dots are hidden 

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅    |
| Android |    ❌     |

## Checklist

- [x] I have tested this on a device and a simulator

### CODE REVIEW

File: ios/ReactNativePageView.m [hunk 1]
```diff
@@ -472,6 +473,17 @@ - (void)shouldShowPageIndicator:(BOOL)showPageIndicator {
     if (_reactPageIndicatorView){
         _reactPageIndicatorView.hidden = !showPageIndicator;
     }
+    
+    // https://github.com/react-native-community/react-native-viewpager/issues/182
+    if (_transitionStyle == UIPageViewControllerTransitionStylePageCurl) {
+        NSArray *subviews = self.reactPageViewController.view.subviews;
+        for (int i=0; i<[subviews count]; i++) {
```
Comment (michalchudziak) [inline ios/ReactNativePageView.m:480]: You can use
```objc
for (id subview in subviews) {
...
}
```

if your min target is iOS 10 :)

File: ios/ReactNativePageView.m [hunk 2]
```diff
@@ -472,6 +473,17 @@ - (void)shouldShowPageIndicator:(BOOL)showPageIndicator {
...
+            if ([[subviews objectAtIndex:i] isKindOfClass:[UIPageControl class]]) {
```
Comment (mdjastrzebski) [inline ios/ReactNativePageView.m:481]: You can use `subviews[i]` instead of `[subviews objectAtIndex:i]` (the same also below).

Comment (troZee) [reply ios/ReactNativePageView.m]: I used https://github.com/react-native-community/react-native-viewpager/pull/184#discussion_r442012212
hence I don't need to use index anymore :D

**Comment by troZee** [comment]
I would like to dig more to find root cause of this issue, hence I hold this PR.

cc @dratwas

**Comment by robban** [comment]
I can confirm that this fixes the issue in my app, thank you 👍

**Comment by troZee** [comment]
@robban thank you for checking this PR. You can use it as temporary workaround using patch-package. I would like to dig more to find better fix.

**Comment by troZee** [comment]
it has been fixed here: https://github.com/react-native-community/react-native-viewpager/pull/189

**Pull request closed without merging**